pkg/internal/server/grpc/validators.go

```diff
@@ -17,42 +17,101 @@ limitations under the License.
 package grpc
 
 import (
+	"context"
+
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/kubernetes-csi/external-snapshot-metadata/pkg/api"
+	snapshot "github.com/kubernetes-csi/external-snapshotter/client/v6/clientset/versioned"
 )
 
-func ValidateGetMetadataAllocatedRequest(req *api.GetMetadataAllocatedRequest) error {
+func (s *Server) ValidateGetMetadataAllocatedRequest(ctx context.Context, req *api.GetMetadataAllocatedRequest) (string, error) {
```

Comment:
I think it is a mistake to overload a validation function to also handle the translation of the snapshot handle.

If we re-organize by handler then we could simply do this in the `newXXXXRequest` structure.